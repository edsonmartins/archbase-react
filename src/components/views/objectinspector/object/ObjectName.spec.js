import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ObjectName } from './ObjectName';

describe('ObjectName', () => {
  it('should render', () => {
    const tree = TestRenderer.create(<ObjectName name="testvalue" />);

    expect(tree).toMatchSnapshot();
  });

  it('should apply dimming if `dimmed` prop is true', () => {
    const tree = TestRenderer.create(<ObjectName name="testvalue" dimmed={true} />);

    expect(tree).toMatchSnapshot();
  });

  it('should not apply dimming if `dimmed` prop is false', () => {
    const tree = TestRenderer.create(<ObjectName name="testvalue" dimmed={false} />);

    expect(tree).toMatchSnapshot();
  });

  it('Accepts and applies additional `style` prop', () => {
    // Test that a custom `style` props is passed and applied to <span/>
    const style = { color: 'hotpink' };
    const tree = TestRenderer.create(<ObjectName styles={style} />);

    expect(tree).toMatchSnapshot();
  });
});
