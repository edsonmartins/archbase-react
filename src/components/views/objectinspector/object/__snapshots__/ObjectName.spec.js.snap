// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectName Accepts and applies additional \`style\` prop 1`] = `
<span
  style={
    Object {
      "color": "hotpink",
    }
  }
/>
`;

exports[`ObjectName should apply dimming if \`dimmed\` prop is true 1`] = `
<span
  style={
    Object {
      "color": "rgb(136, 19, 145)",
      "opacity": 0.6,
    }
  }
>
  testvalue
</span>
`;

exports[`ObjectName should not apply dimming if \`dimmed\` prop is false 1`] = `
<span
  style={
    Object {
      "color": "rgb(136, 19, 145)",
    }
  }
>
  testvalue
</span>
`;

exports[`ObjectName should render 1`] = `
<span
  style={
    Object {
      "color": "rgb(136, 19, 145)",
    }
  }
>
  testvalue
</span>
`;
